VDA-DID-Registry/contracts/VeridaDIDRegistry.sol

```diff
@@ -5,419 +5,163 @@ pragma solidity ^0.8.6;
-       return owner;
-     }
-     return identity;
-  }
+  mapping(address => uint) private _nonce;
 
   /**
-   * @notice Check wheter delegate is valid.
-   * @param identity - DID that registered delegate
-   * @param delegateType - delegate type in bytes32 format
-   * @param delegate - delegate to check
-   * @return boolean - result
+   * @notice Map of endpoints
+   * @dev DID address => List of endpoints
    */
-  function validDelegate(address identity, bytes32 delegateType, address delegate) external view returns(bool) {
-    uint validity = delegates[identity][keccak256(abi.encode(delegateType))][delegate];
-    return (validity > block.timestamp);  
-  }
+  mapping (address => EnumerableSet.StringSet) private _endpoints;
 
   /**
-   * @notice Change owner of DID
-   * @dev Only called after checking the transaction signature
-   * @param identity - DID
-   * @param newOwner - new owner address
+   * @notice Initialize
    */
-  function _changeOwner(address identity, address newOwner) internal {
-    owners[identity] = newOwner;
-    emit DIDOwnerChanged(identity, newOwner, changed[identity]);
-    changed[identity] = block.number;
+  function initialize() public initializer {
+    __Ownable_init();
   }
-
+  
   /**
-   * @notice Change owner of DID
-   * @dev Check transaction signature and call internal function
-   * @param identity - DID
-   * @param newOwner - new owner address
-   * @param signature - transaction signature
+   * @dev See {IVeridaDIDRegistry}
    */
-  function changeOwner(address identity, address newOwner, bytes calldata signature) external {
+  function register(address didAddress, string[] calldata endpoints, bytes calldata signature ) external override {
     {
-      bytes memory unsignedMsg = abi.encodePacked(
-        identity,
-        newOwner,
-        nonce[identity]
-      );
-      require(VeridaDataVerificationLib.validateSignature(unsignedMsg, signature, identityOwner(identity)), "Invalid Signature");
-      nonce[identity]++;
-    }
-    _changeOwner(identity, newOwner);
-  }
+      bytes memory rawMsg = abi.encodePacked(didAddress, "/");
+      for (uint i = 0; i < endpoints.length; i++) {
+        rawMsg = abi.encodePacked(rawMsg, endpoints[i], "/");
+      }
 
-  /**
-   * @notice Add a delegate to DID
-   * @dev Only called after checking the transaction signature
-   * @param identity - DID
-   * @param delegateType - delegate type in bytes32 format
-   * @param delegate - delegate to check
-   * @param validity - valid duration of delegate
-   */
-  function _addDelegate(address identity, bytes32 delegateType, address delegate, uint validity) internal {
-    delegates[identity][keccak256(abi.encode(delegateType))][delegate] = block.timestamp + validity;
-    emit DIDDelegateChanged(identity, delegateType, delegate, block.timestamp + validity, changed[identity]);
-    changed[identity] = block.number;
-  }
+      rawMsg = abi.encodePacked(rawMsg, _nonce[didAddress]);
```

Comment:
Won't nonce always be 0 when calling register?

Reply:
No need.
What happend user revoke & register again? That time, nonce will not be 0.

Reply:
Once it is revoked, they can't re-register the same DID.

Reply:
Updated for revoked DIDs. By the way, we should not check whether nonce equals to 0. Because, `register()` function updates endpoints for already registered DIDs.

Reply:
@ITStar10 When a DID is revoked it can no longer be registered. From reading the code, it appears I can re-register a DID after it's revoked.

Can you add a unit test to show this works as expected?

Reply:
Resolved in the latest commit.